java/src/jmri/InstanceManager.java

```diff
@@ -92,13 +92,8 @@
      * @param type The class Object for the item's type. This will be used as
      *             the key to retrieve the object later.
      */
-    static public <T> void store(@Nonnull T item, @Nonnull Class<T> type) {
+    public static <T> void store(@Nonnull T item, @Nonnull Class<T> type) {
         log.debug("Store item of type {}", type.getName());
-        if (item == null) {
```

Comment:
I know SonarLint thinks this is not needed given the annotation, but since these annotations are not enforced at run time, you must throw the NPE now to ensure later access does not have issues.

Reply:
It isn't just SonarLint that thinks they are unnecessary when the parameter values are annotated to indicate they can't be null.

Basically, the annotation becomes part of the contract for the method, and users of the method SHOULD follow that contract.

I'll put the null checks back in for now.